articles/quickstart/native/ios-swift-siwa/_includes/_login_centralized.md

```diff
@@ -0,0 +1,306 @@
+<!-- markdownlint-disable MD002 MD041 -->
+
+<%= include('../../_includes/_getting_started', { library: 'Swift' }) %>
+
+Add your credentials in the `Auth0.plist` file. If the file does not exist in your project yet, create it:
+
+```xml
+<!-- Auth0.plist -->
+
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+  <key>ClientId</key>
+  <string>${account.clientId}</string>
+  <key>Domain</key>
+  <string>${account.namespace}</string>
+</dict>
+</plist>
+```
+
+<%= include('../../_includes/_ios_dependency_centralized') %>
+
+## Add the "Sign In With Apple" Button
+
+As an example of adding the sign-in button to your view, add a `UIStackView` to your main storyboard and create an outlet in `ViewController.swift`:
+
+```swift
+ @IBOutlet weak var loginProviderStackView: UIStackView!
+```
+
+Next, add the `AuthenticationServices` Framework to your `ViewController.swift`:
+
+```swift
+import AuthenticationServices
+```
+
+Add a function that attaches a `ASAuthorizationAppleIDButton` button to the stack view:
+
+```swift
+func setupProviderLoginView() {
+  // Create Button
+  let authorizationButton = ASAuthorizationAppleIDButton()
+
+  // Add Callback on Touch
+  authorizationButton.addTarget(self, action: #selector(handleAuthorizationAppleIDButtonPress), for: .touchUpInside)
+
+  //Add button to the UIStackView
+  self.loginProviderStackView.addArrangedSubview(authorizationButton)
+}
+```
+
+Call this function in `viewDidLoad`:
+
+```swift
+override func viewDidLoad() {
+    super.viewDidLoad()
+
+    setupProviderLoginView()
+}
+```
+
+## Handle Authorization
+
+The following code shows the implementation of a button handler that initializes an authorization request and launches
+the native Sign In With Apple flow:
+
+```swift
+@objc
+func handleAuthorizationAppleIDButtonPress() {
+    // Create the authorization request
+    let request = ASAuthorizationAppleIDProvider().createRequest()
+
+    // Set scopes
+    request.requestedScopes = [.email, .fullName]
+
+    // Setup a controller to display the authorization flow
+    let controller = ASAuthorizationController(authorizationRequests: [request])
+
+    // Set delegates to handle the flow response.
+    controller.delegate = self
+    controller.presentationContextProvider = self
+
+    // Action
+    controller.performRequests()
+}
+```
+
+:::note
+At this time the only scopes that Apple support are `email` and `fullName`. For more information, see [ASAuthorization.scope](https://developer.apple.com/documentation/authenticationservices/asauthorization/scope).
```

Comment:
I believe we were recommending only email?  I can't recall the Slack thread however where that was brought up.

Reply:
Pinged you on Slack but I managed to verify that fullName does actually work, at least in the simulator.